### PR TITLE
Improve dangling reference example and move to its own slide

### PR DESCRIPTION
**Pull request by randomPoison**
The current example demonstrating how rustc prevents dangling references is really gross and hard to read (my own fault lol, I wrote that example). I finally realized that there's a much simpler, easier to read way of expressing the same thing. I also moved this to its own slide after the reference slides so that we can call it out as an early example of the borrow checker. I then call back to this example in the borrow checker slide to remind students that the aliasing rule isn't the only thing the borrow checker is enforcing.

### CODE REVIEW

File: src/references/dangling.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+to.
+
+<!-- mdbook-xgettext: skip -->
+
+```rust,editable,compile_fail
+fn main() {
+    let x_ref = {
+        let x = 10;
+        &x
+    };
```
[the remark this answers is not in the document]
Comment (randomPoison) [reply src/references/dangling.md:18]: That's a good point. I always call out rust's scoping rules here for that reason, using this as an opportunity to remind students Rust may work differently if they come from e.g. python. I'll add a speaker note about it.

**Comment by randomPoison** [comment]
> One nitpick here is that it might be better to name the slide/section after a positive notion like "Reference Validity" instead of after something that Rust forbids ("Dangling References").

Good point, I'll change the name.

> In the list of slides, we have a list of different types Rust has under the "References" section, and we're inserting "Dangling References" into that list despite it being a very different kind of thing.

Hm, that's a good point. Do you think it'd make sense to put it at the end of the section instead, then? I think that'd work because we can then call out at all of the different flavors of reference have this property.

**Comment by randomPoison** [comment]
It occurs to me that now that the slide is focused on "reference validity", it would make sense to also mention that references cannot be `null` in Rust. We're kind of glossing over that currently (I usually mention it, but it's not called out anywhere in the slides). I think it fits in nicely now that we have a slide dedicated to reference validity.

**Review by djmitche — APPROVED** [review]
Agreed, that's a good thing to call out!

**Pull request merged**